torchmultimodal/modules/encoders/swin_transformer_3d_encoder.py

```diff
@@ -0,0 +1,518 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Modified from 2d Swin Transformers in torchvision:
+# https://github.com/pytorch/vision/blob/main/torchvision/models/swin_transformer.py
+
+from functools import lru_cache, partial
+from typing import Callable, List, Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import nn, Tensor
+from torchvision.models.vision_transformer import MLPBlock
```

Comment:
Is it possible to bootstrap from our MLP layer?

https://github.com/facebookresearch/multimodal/blob/main/torchmultimodal/modules/layers/mlp.py#L13

Reply:
I think we can do this, but take note that there is a slight different.
The original MLP: https://github.com/facebookresearch/omnivore/blob/main/models/swin_transformer_3d.py#L65
and the one in torchvision have final drop layer after at the end. While the torchmultimodal MLP layer can't have dropout layer at the end.

Not sure how big is the effect of this dropout, but we could try.

Reply:
If the end state of this module is to upstream to torchvision, I think it's fine to stick with vision's components here. Also I notice that `MLPBlock` in torchvision is cusotmized for transformer so I don't think our MLP is the equivalent.